src/components/hint/index.tsx

```diff
@@ -572,8 +583,10 @@ class Hint extends Component<HintProps, HintState> {
             {this.renderHintContainer()}
           </Modal>
         ) : (
-          // this.renderOverlay(),
-          this.renderHintContainer()
+          <>
+            {onBackgroundPress && this.renderOverlay()}
```

Comment:
No need to condition this use case with `onBackgroundPress` 
We take care of it in `renderOverlay` function, where we render the actual overlay. 
Technically, the user can still have a hint with a background overlay